core/src/main/java/org/jivesoftware/LoginDialog.java

```diff
@@ -273,16 +275,22 @@ protected XMPPTCPConnectionConfiguration retrieveConnectionConfiguration() {
             builder.setProxyInfo( proxyInfo );
         }
 
-        if (useSSL) {
+        if ( securityMode != ConnectionConfiguration.SecurityMode.disabled && useOldSSL ) {
             if (!hostPortConfigured) {
+                // SMACK 4.1.9 does not support XEP-0368, and does not apply a port change, if the host is not changed too.
+                // Here, we force the host to be set (by doing a DNS lookup), and force the port to 5223 (which is the
+                // default 'old-style' SSL port).
+                builder.setHost( DNSUtil.resolveXMPPDomain( loginServer, null ).get( 0 ).getFQDN() );
                 builder.setPort( 5223 );
             }
             builder.setSocketFactory( new DummySSLSocketFactory() );
+            // SMACK 4.1.9  does not recognize an 'old-style' SSL socket as being secure, which will cause a failure when
+            // the 'required' Security Mode is defined. Here, we work around this by replacing that security mode with an
+            // 'if-possible' setting.
+            builder.setSecurityMode( ConnectionConfiguration.SecurityMode.ifpossible );
```

Comment:
It's not an ideal solution, I agree. But, when using 'old-style SSL', we cannot set security mode "required" (which is what you'd ideally want, I think). 

We cannot do this, because Smack (at least in version 4.1.9) will not detect that the connection is encrypted when using 'old-style SSL'. If you set the mode to 'required', Smack will throw an error (as it didn't detect encryption, but knows that encryption is required).